packages/synapse-react-client/src/components/Plot/DotPlot.tsx

```diff
@@ -83,24 +95,80 @@ function getLayout(
 function createArrayOfGroupValues(
   headers: string[],
   items: GraphItem[],
-): any[] {
-  const result = new Array(headers.length)
+): number[] {
+  const result: number[] = new Array(headers.length)
   items.forEach(item => {
     const index = headers.indexOf(item.y)
-    result[index] = item.x
+    if (index !== -1) {
+      result[index] = item.x
+    }
```

Comment:
fix case where items that did not exist in the array were added with an index of `-1`